### ISSUE TITLE
Regression 1.19 -> 1.22.1

### ISSUE THREAD

**Issue by Flow86** [c-ares/c-ares#640]
Hi,

I'm doing this on a windows and/or linux machine amd64/x86:

```
const unsigned short dns_server_port = 53;

ares_channel channel;
ares_options options;
int optmask = 0;

options.udp_port = dns_server_port;
optmask |= ARES_OPT_UDP_PORT;

ares_init_options(&channel, &options, optmask);

// ares_query etc ...
```

When running a query with 1.22.1, I get a timeout.

Wireshark reveiled me this:
```
Internet Protocol Version 4, Src: 192.168.x.x, Dst: 8.8.8.8
User Datagram Protocol, Src Port: 62216, Dst Port: 13568
```

Dst Port 13568? The port is set in network byte order?!?

With 1.19 I correctly get an answer from the DNS server back:

```
Internet Protocol Version 4, Src: 192.168.x.x, Dst: 8.8.8.8
User Datagram Protocol, Src Port: 53501, Dst Port: 53
```

In the manual its stated that the port should be provided in host byte order:

> The port to use for queries over UDP, in host byte order. The default value is 53, the standard name service port. 

Somewhere somehow it looks like there is a `htons` to often in c-ares in the mean time?
(Simplified the example of course, since you could question "why to add a custom port if port 53 is used anyway, this is only for the example)

**Comment by bradh352** [comment]
Odd, we have test cases in the suite for this as we generate a fake DNS server listening on a random port, so I'd have thought that would catch this.  I'll check it out now.

**Comment by Flow86** [comment]
adding a `htons` manually

like so:
```
options.udp_port = htons(dns_server_port);
```

does the trick for now,

but it looks like the test didnt catch at least one case :-)

oh, I also set a nameserver to query:

```
ares_addr_node* servers = nullptr;

ares_addr_node* server_entry = alloc_addr_node();
ares_inet_pton(AF_INET, "8.8.8.8", &server_entry->addr.addr4);
server_entry->family = AF_INET;
append_addr_list(server_list, server_entry);

optmask |= ARES_OPT_SERVERS;
ares_set_servers(channel, servers);
```

perhaps it has something to do with setting a server explicitly? 

while looking into the c-ares code, my quick guess would be something around `ares__sconfig_get_port` vs `ares__open_connection` - where again a `htons` call is done

if I do not set one, it does work correctly too (querying local resolver)

**Comment by bradh352** [comment]
fixed in fb52c3f

**Comment by Flow86** [comment]
Tested it here, works. Thank you.
